Hig-182 logging

brianhlee opened the pull request:
ethan already had all the log statements written out. Just converted everything using json-logger and added username and timestamps

Commit:
Higs-182 logging

ethan already had all the log statements written out. Just converted everything using json-logger and added username and timestamps

brianhlee commented:
pcf.log is just an example output after running the s3 examples

ethanwlo commented on pcf/core/particle.py:
```diff
@@ -11,16 +11,36 @@
+
+# create console handler and set level to debug
+ch = logging.FileHandler(filename)
+ch.setLevel(logging.DEBUG)
```
So I assume this json formatter only applies if user (Particles that inherit from this class) sets log level to debug?

ethanwlo commented on pcf/core/particle.py:
```diff
@@ -11,16 +11,36 @@
+ch.setLevel(logging.DEBUG)
+
+# create formatter
+formatter = jsonlogger.JsonFormatter('%(asctime)s - %(name)s - %(funcName)s - %(levelname)s - %(message)s')
```
%(user)s is more descriptive than %(name)s

ethanwlo reviewed (commented):
In general PR is assuming it's more detailed logging for debug mode. But we want to looking into this as more of a mechanism that will ultimately enable auditing. You can add a formatter for INFO, what an "auditor" wants to know is who attempts to deploy/terminate what, at what time, and if it was successful or not.

brianhlee committed:
Merge branch 'develop' into logging

Commit:
added (module)s to the formatter

moved the logger into the class so that particles can call it using self.logger

Commit:
Merge branch 'logging' of https://github.com/bhl5kj/Particle-Cloud-Framework into logging

brianhlee committed:
Merge branch 'develop' into logging

anovis commented on requirements.txt:
```diff
@@ -17,6 +17,7 @@ commentjson==0.7.1
 botocore==1.12.72
 deepdiff==3.3.0
 placebo==0.8.2
+python-json-logger==0.1.10
```
lets not add a dependency. should be easy to just create a custom formatter.

anovis commented on pcf/core/particle.py:
```diff
@@ -44,6 +44,27 @@ class Particle(object, metaclass=MetaParticle):
+
+    # create formatter
+    formatter = jsonlogger.JsonFormatter(
+        '%(asctime)s - %(name)s - %(module)s - %(funcName)s - %(levelname)s - %(message)s')
```
might want to have been json messaging. this would require changing the inputs to the info and debug calls. 

for example:
```logger.info"{0}: setting desired state to {1}".format(self.pcf_id, self.desired_state)```

to 
```logger.info"{msg:msg, pcf_id:id,state:state, desired_state:{}")```

anovis commented on pcf/core/particle.py:
```diff
@@ -44,6 +44,27 @@ class Particle(object, metaclass=MetaParticle):
+
+    # create formatter
+    formatter = jsonlogger.JsonFormatter(
+        '%(asctime)s - %(name)s - %(module)s - %(funcName)s - %(levelname)s - %(message)s')
```
this doesnt cvt messages to json. just adds extra fields such as time, name, level.

brianhlee replied:
that's what LogRecord does. JsonFormatter is built on top of it to turn it into json
